warp-tls/Network/Wai/Handler/WarpTLS.hs

```diff
@@ -176,6 +176,8 @@ tlsSettingsRef cert key =
         { certSettings = CertFromRef cert [] key
         }
 
+{-# DEPRECATED tlsSettingsRef "Will be removed in the next major release" #-}
```

Comment:
I'd like to read _why_ something is deprecated when I get warnings and possibly also what to do instead of the deprecated functionality.

Could the reason that this function is deprecated be added, at least?
(Same with the other change)

Reply:
Indeed. But I think that @tmcl or @vdukhovni are better to write up the background than me.